st2-ask/notification_rule.yaml

```diff
@@ -0,0 +1,33 @@
+---
+name: "notify_slack_execution_change_managers"
+pack: "st2"
+description: Watches for an execution status of "pending" for a certain action ref
+enabled: true
+
+trigger:
+  type: core.st2.generic.actiontrigger
+
+criteria:
+  trigger.action_ref:
+      type: equals
+      pattern: st2.ask
+  trigger.status:
+      type: equals
+      pattern: pending
+
+  # In order to control who to notify, the "tag" parameter for the st2.ask action would be
+  # used to indicate metadata about this specific instance (would be set in the workflow).
+  # Since it's in the parameters, we can get a handle on it here in the rule and use it to
+  # inform our decision on who to notify.
+  #
+  # This is a very static, simple example - and would require one rule for each notification "group",
+  # or type of st2.ask action. See notification_rule_workflow.yaml for a more dynamic example
+  trigger.parameters.tag:
+      type: equals
+      pattern: managers
+
+action:
+  ref: slack.post_message
+  parameters:
+    channel: "#managers"
```

Comment:
Because the only way of mapping tags to various notification logic would be available in a workflow, not in a rule. If you wanted to do it purely with rules, you'd have to either have one rule per tag, or maybe play around with st2kv, but as you allude, this isn't much better. It would be preferable just to pass this value into a workflow that manages the complexity of how the user's specific organization chooses to do notifications.

I can put together an example, but that's the english version.